AssetManagementTestProject/TestCases/LoginTest.cs

```diff
@@ -7,27 +7,49 @@ namespace AssetManagementTestProject;
 [TestFixture]
 public class LoginTest : NUnitWebTestSetup
 {
+    protected ChangePassword1stTimePage ChangePw1stTime;
+
+    [SetUp]
+    public void SetUp()
+    {
+        ChangePw1stTime = new ChangePassword1stTimePage();
+    }
     [Test]
     public void UserLoginSuccess()
     {
-        LoginPage.Login(Constant.ADMIN_USERNAME_HN, Constant.BASE_ADMIN_PASSWORD);
-        Asserter.AssertStringEquals(HomePage.ReturnHomePageUrl(), Constant.BASE_URL);
+        LoginPage.Login(Constant.ADMIN_USERNAME_HN, Constant.ADMIN_PASSWORD_HN);
+        DriverBaseAction.WaitToBeVisible(HomePage.ReturnHeaderHomePage());
+        DriverBaseAction.CompareUrls(Constant.BASE_URL);
     }
     [Test]
     public void UserLoginFirstTime()
     {
-        LoginPage.Login(Constant.STAFF_USERNAME_1, Constant.BASE_STAFF_PASSWORD_1);
+        LoginPage.Login(Constant.STAFF_USERNAME_1ST_TIME, Constant.STAFF_PASSWORD_1ST_TIME);
         DriverBaseAction.IsElementDisplayed(ChangePw1stTime.AskChangePwFirstLogin());
         DriverBaseAction.CompareUrls(ChangePw1stTime.ReturnExpectedChangePw1stTimeUrl());
     }
-    [TestCase(Constant.ADMIN_USERNAME_HN, Constant.BASE_ADMIN_PASSWORD)]
-    public void UserCanLoginToTheApp(string userName, string password)
+    [Test] 
+    public void UserLoginWithNewPassword()
+    {
+        LoginPage.Login(Constant.ADMIN_USERNAME, Constant.CHANGED_ADMIN_PASSWORD);
+        DriverBaseAction.WaitToBeVisible(HomePage.ReturnHeaderHomePage());
+        DriverBaseAction.CompareUrls(Constant.BASE_URL);
+    }
+    [Test]
+    public void AdminCanLoginToTheApp()
     {
-        LoginPage.Login(userName, password);
-        Asserter.AssertStringEquals(HomePage.ReturnHomePageUrl(), Constant.BASE_URL);
+        LoginPage.Login(Constant.ADMIN_USERNAME_HN, Constant.ADMIN_PASSWORD_HN);
         DriverBaseAction.AreElementsDisplayed(MenuBarLeft.ReturnMenuBar());
-        HomePage.SelectLogout();
-        LogoutPopup.LogOutOfPage();
+    }
+    [Test]
+    public void StaffCanLoginToTheApp()
+    {
+        LoginPage.Login(Constant.STAFF_USERNAME, Constant.STAFF_PASSWORD);
+        DriverBaseAction.IsElementDisplayed(MenuBarLeft.ReturnHomeBtn());
```

Comment:
This is a meaningless step.
The expectation is a clause that defines what behavior will happen next. DriverBaseAction.IsElementDisplayed() was not a behavior
Then it should not be an expectation. So the reader cannot understand what are you trying to do.
You have to do the assert function such as: ....Element(...).IsDisplayed()
or ....ElementIsDisplayed(....)

Reply:
I think while this is possible, but this means using an assert function outside WebDriverAction? Or did you mean to rewrite the method name from IsElementDisplayed to ElementIsDisplayed?

Reply:
Done. I added a method in the `Asserter` class to verify whether element(s) is displayed